[138] Change_password_for_registered_user - Implement password reset

RockTF opened the pull request:
[[138] - Change_password_for_registered_user - Implement password reset
](https://github.com/OlhenShu/ProjectStage_Team1_May/issues/138)

- [x] Obtained approval from the owner
- [x] Made changes to the code
- [x] Requested a review

RockTF committed:
issues/356 - fixed UserRepo (a lot more changes need to be made! But you will see all this in the following tasks)

"countOfMutualFriends"
"countOfMutualFriends"
"getStatusUser"

tellmant committed:
[357] fixed response status code for /user/deleteProfilePicture endpoint from 401 to 200

tellmant committed:
[357] updated pom.xml with spring-security-test and spring-boot-starter-test dependencies

tellmant committed:
[357] add tests cases for UserController to test /user/deleteProfilePicture endpoint

tellmant committed:
[352] fixed response status code for /user endpoint from 200 to 403

tellmant committed:
[352] fixed getUserByPrincipal() to get getUserUpdateDtoByEmail() with authorized user email

tellmant committed:
[352] updated pom.xml with spring-security-test and spring-boot-starter-test dependencies

tellmant committed:
[352] add test cases for UserController to test /user endpoint

tellmant committed:
[289] fixed response status code for /user/profilePicture endpoint from 500 to 200

tellmant committed:
[289] updated pom.xml with spring-security-test and spring-boot-starter-test dependencies

ushh789 committed:
[215] added verification if user exists in sendUserViolationEmail()

ushh789 committed:
[215] added tests for sendUserViolationEmail()

tellmant committed:
[289] add test cases for UserController to test /user/profilePicture endpoint

ushh789 committed:
[246] fixed tests for sendUserViolationEmail()

nataliiapovzhyk committed:
added validation for email in Auther

nataliiapovzhyk committed:
created new table with user_friends

nataliiapovzhyk committed:
db drop

nataliiapovzhyk committed:
properties

df7734 committed:
[74]-add initial subscribe build

nataliiapovzhyk committed:
update liquibase with user friends table

nataliiapovzhyk committed:
added permisions for friends

nataliiapovzhyk committed:
allow only authorized users for friends GET

nataliiapovzhyk committed:
revert settings for ddl-auto

nataliiapovzhyk committed:
added liquibase

df7734 committed:
[74]-add NewsSubscriber entity

df7734 committed:
[74]-add NewsSubscriber entity & other minor changes

df7734 committed:
[74]-add Controller Test

nataliiapovzhyk committed:
new functionality to get all user friends

df7734 committed:
[74]-state of feature: NewsSubscriber added to db via MVP changelog. Subscribe and GetAll works with swagger

df7734 committed:
[74]-add unsubscribe

df7734 committed:
[74] unsubscribe realised

tellmant committed:
[106] updated pom.xml with mapstruct and mapstruct-processor dependencies

nataliiapovzhyk committed:
friend controller test

nataliiapovzhyk committed:
remover liquibase

nataliiapovzhyk committed:
remove change for modelUtils

nataliiapovzhyk committed:
User to FriendDto mapper test

nataliiapovzhyk committed:
mapper test

nataliiapovzhyk committed:
merge dev into feature/ecoFriends

nataliiapovzhyk committed:
fix text

df7734 committed:
[74]-minimum viable feature

tellmant committed:
[106] add FriendDto to display info about user's friend

tellmant committed:
[106] add FriendDtoMapper to convert User entity to FriendDto

tellmant committed:
[106] updated SecurityConfig to secure friends related endpoints

tellmant committed:
[106] add methods to get information about user's friends

tellmant committed:
[106] add additional methods to request user's friends lists from database

tellmant committed:
[106] add additional methods to UserService related to user's friends lists

tellmant committed:
[106] implemented methods to process requested information about user's friends

tellmant committed:
[106] add Swagger docs to UserController undocumented methods

IliaWithHat committed:
Fix test

df7734 committed:
[74] unsubscribe link inside email fixed

nataliiapovzhyk committed:
Delete friends end point and methods

nataliiapovzhyk committed:
tests for delete friend

df7734 committed:
[74] Minor changes

df7734 committed:
Merge branch 'dev' of https://github.com/GreenCityProject/GreenCityUser21_team_1_May into features/74-Receive_interesting_news

df7734 committed:
[74] minor changes

nataliiapovzhyk committed:
update of user friends table name

nataliiapovzhyk committed:
Merge branch 'feature/105-106_Eco_Friends_API' into feature/107_search_and_add_friend

nataliiapovzhyk committed:
remove of commented lines

nataliiapovzhyk committed:
merge of branch feature/106-View_my_All_Eco_Friends_list

nataliiapovzhyk committed:
updated to use FriendDTO

nataliiapovzhyk committed:
remove change for other task

nataliiapovzhyk committed:
remove not needed test

ushh789 committed:
[246] flipped if-else in sendUserViolationEmail()

df7734 committed:
[74] add unsubscribe response

RockTF committed:
138_Change_password_for_registered_user - Implement password reset functionality

Added the ability for users to reset their passwords for the OwnSecurity service. This includes new validation checks for the reset process and accompanying unit tests to ensure functionality correctness. Also updated the security configuration and respective controllers to handle the password reset endpoints.

RockTF committed:
138_Change_password_for_registered_user - Implement password reset functionality

Added the ability for users to reset their passwords for the OwnSecurity service. This includes new validation checks for the reset process and accompanying unit tests to ensure functionality correctness. Also updated the security configuration and respective controllers to handle the password reset endpoints.

RockTF committed:
Update password reset methods with new DTO

This commit updates the OwnSecurityService and OwnSecurityController classes to use the ResetPasswordDto instead of the previously used SetPasswordDto. Also, a typo in the reset password method name in OwnSecurityController has been fixed.

RockTF committed:
Update password reset methods with new DTO

This commit updates the OwnSecurityService and OwnSecurityController classes to use the ResetPasswordDto instead of the previously used SetPasswordDto. Also, a typo in the reset password method name in OwnSecurityController has been fixed.

tellmant committed:
[106] updated file structure for Eco Friends API

tellmant committed:
Merge branch 'dev' into bug/289-PATCH_user_profilePicture_is_500_need_200

tellmant committed:
Merge pull request #24 from GreenCityProject/bug/289-PATCH_user_profilePicture_is_500_need_200

[289] fixed response status code for /user/profilePicture endpoint from 500 to 200

tellmant committed:
Merge branch 'dev' into bug/357-PATCH_user_deleteProfilePicture_is_401_need_200

tellmant committed:
Merge pull request #14 from GreenCityProject/bug/357-PATCH_user_deleteProfilePicture_is_401_need_200

[357] fixed response status code for /user/deleteProfilePicture endpoint from 401 to 200

tellmant committed:
Merge branch 'dev' into bug/352-GET_user_userDto_is_200_need_403

tellmant committed:
Merge pull request #20 from GreenCityProject/bug/352-GET_user_userDto_is_200_need_403

[352] fixed response status code for /user endpoint from 200 to 403

RockTF committed:
Removed update password feature, modified reset password feature

The code removed the update password feature as it was unwanted. Also, the reset password feature was modified to reset the password if the new password and the confirm password are equal and the old password provided is equal to the existing password. Finally, the endpoint route was also changed from "reset-password" to "changePassword".

RockTF committed:
Removed update password feature, modified reset password feature

The code removed the update password feature as it was unwanted. Also, the reset password feature was modified to reset the password if the new password and the confirm password are equal and the old password provided is equal to the existing password. Finally, the endpoint route was also changed from "reset-password" to "changePassword".

RockTF committed:
Renamed resetPassword method to updateCurrentPassword

The method for password change has been renamed from "resetPassword" to "updateCurrentPassword" throughout the application, including in tests and controllers, for better semantics and clarity.

RockTF committed:
Replace ResetPasswordDto with UpdatePasswordDto in OwnSecurityService

Renamed ResetPasswordDto to UpdatePasswordDto and updated all its references in OwnSecurityService and related test classes. Furthermore, added a new exception handler for PasswordSameAsOldException.

RockTF committed:
Replace ResetPasswordDto with UpdatePasswordDto in OwnSecurityService

Renamed ResetPasswordDto to UpdatePasswordDto and updated all its references in OwnSecurityService and related test classes. Furthermore, added a new exception handler for PasswordSameAsOldException.

df7734 committed:
[74] change security config subscriber/sendEcoNews

ushh789 committed:
Merge pull request #25 from GreenCityProject/bug/246-sendUserViolation_404_response

[246] fixed sendUserViolation() with wrong email Response 404 Not found

IliaWithHat committed:
Merge pull request #40 from GreenCityProject/bug/fix_test

Fix test

nataliiapovzhyk committed:
updated friends table name

nataliiapovzhyk committed:
updated friends table name

nataliiapovzhyk committed:
Update application-docker.properties

nataliiapovzhyk committed:
Update application-dev.properties

nataliiapovzhyk committed:
Merge pull request #43 from GreenCityProject/feature/105-106_Eco_Friends_API

[105][106] Eco Friends API

nataliiapovzhyk committed:
fix missing coma

nataliiapovzhyk committed:
fix missing coma

nataliiapovzhyk committed:
Merge branch 'dev' into feature/108_delete_friend

df7734 committed:
[74] email removed from unsub link to except malicious use

df7734 committed:
[74] removed unused import

tellmant commented on core/src/main/java/greencity/security/controller/OwnSecurityController.java:
It would be better to change the response status for password reset from `HttpStatus.CREATED (201)` to `HttpStatus.OK (200)` for the `updateCurrentPassword` endpoint.

RockTF replied:
This is the correct answer for a Post request)

RockTF replied:
But you are right about the 500 status. It seems I didn't add a handler for the custom exception in our error handler.

tellmant reviewed (changes requested):
## Recommended changes:
1. Change response status for the **updateCurrentPassword** endpoint from **201 (Created)** to **200 (Ok)**

## Required changes:
1. Handle the case where a user attempts to enter the same password as the old one. Currently, this results in an **Internal Server Error (500)**, which should be handled properly.
   > *Example:*
   >
   > ![image](https://github.com/GreenCityProject/GreenCityUser21_team_1_May/assets/128901331/1bc670a6-c925-4b62-9686-9413587ef404)

IliaWithHat committed:
Merge pull request #45 from GreenCityProject/feature/108_delete_friend

[108] - fixing missing coma!!! - Feature/108 delete friend

RockTF committed:
Merge remote-tracking branch 'origin/138_Change_password_for_registered_user' into 138_Change_password_for_registered_user

# Conflicts:
#	core/src/main/java/greencity/security/controller/OwnSecurityController.java
#	service-api/src/main/java/greencity/security/service/OwnSecurityService.java
#	service/src/main/java/greencity/security/service/OwnSecurityServiceImpl.java

RockTF committed:
Refactor code for improved readability

Indents and whitespaces are streamlined across modules for better uniformity and readability. This cosmetic commit does not introduce functional changes to the application but aims to improve code maintainability.

RockTF committed:
Refactor code for improved readability

Indents and whitespaces are streamlined across modules for better uniformity and readability. This cosmetic commit does not introduce functional changes to the application but aims to improve code maintainability.

RockTF committed:
Merge pull request #9 from GreenCityProject/issues/356

[356] - fixed UserRepo

RockTF committed:
Add changePassword to post request matchers

The security configuration has been updated to include the "changePassword" endpoint in the post request matchers. Previously, it was mistakenly classified under the "PUT" request matchers. This rectifies the categorization mistake and should ensure the endpoint functions as expected.

df7734 committed:
[74] updated dto and template for email message

df7734 committed:
[74] minor changes in template

df7734 committed:
[74] fixed subscribe response

RockTF committed:
Rearrange password validation checks in OwnSecurityServiceImpl

The order of password validation checks in the `ensureNewPasswordIsValid` method has been changed. Now, the method first checks if the new password and confirmation password are identical before checking if the new password is the same as the old one. This provides a more logical flow of the password validation process.

RockTF committed:
Rearrange password validation checks in OwnSecurityServiceImpl

The order of password validation checks in the `ensureNewPasswordIsValid` method has been changed. Now, the method first checks if the new password and confirmation password are identical before checking if the new password is the same as the old one. This provides a more logical flow of the password validation process.

df7734 committed:
Merge branch 'dev' into features/74-Receive_interesting_news

df7734 committed:
Merge pull request #41 from GreenCityProject/features/74-Receive_interesting_news

[74] Receive interesting news via email

RockTF committed:
Merge remote-tracking branch 'origin/138_Change_password_for_registered_user' into 138_Change_password_for_registered_user

# Conflicts:
#	core/src/main/java/greencity/config/SecurityConfig.java
#	core/src/main/java/greencity/controller/UserController.java
#	core/src/main/java/greencity/exception/handler/CustomExceptionHandler.java
